### DISCUSSION TITLE
Icon creation Chakra v1 vs v0.8

### DISCUSSION THREAD

**Discussion by junaid33** [chakra-ui/chakra-ui#2750] [Q&A] [answered]
Hello,

We are moving to Chakra v1 but I'm confused about how to move our icons over. Our custom icons were defined in theme.js like so:

```
const theme = {
  ...chakraTheme,
  icons: {
    ...chakraTheme.icons,
    menu: {
      path: (
        <path
          fill="currentColor"
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      ),
    },
};

export default theme;
```
We could then reference it like so `<Icon name="menu" />`. The name field is not available on Chakra 1. How can we keep our icons on Theme.js and reference them? Or should we create each icon using createIcon function and import the icon every time?

**Comment by chr-ge** [comment, marked as answer]
Chakra provides two methods for creating your custom icons:
- Using the [Icon](https://chakra-ui.com/docs/components/icon#using-the-icon-component) component
- Using the [createIcon](https://chakra-ui.com/docs/components/icon#using-the-createicon-function) function

**Comment by nerdess** [comment]
I am facing the same question/problem. Referencing an icon using name="foo" was so easy, having to first generate each `<Icon>` and then import it seems so cubersome.

**Reply by ljosberinn** [reply]
The v0 version led to a bloated bundle sadly - because the bundler doesn't know at bundling time which icons are going to be used, _all_ had to be imported, regardless whether you used a single one, none, or actually all. And adding ~20kB for unused icons is hard to justify.
